File: rust/src/leetcode/Q55_jump_game.rs
/*
 * @lc app=leetcode id=55 lang=rust
 *
 * [55] Jump Game
 */

// @lc code=start
impl Solution {
    pub fn can_jump(nums: Vec<i32>) -> bool {
        let mut jump: usize = nums.len() - 1;

        for i in (0..nums.len() - 1).rev() {
            if nums[i] as usize + i >= jump {
                jump = i
            }
        }

        jump == 0
    }
}
// @lc code=end

File: python/Q55_jump_game.py
#
# @lc app=leetcode id=55 lang=python3
#
# [55] Jump Game
#


# @lc code=start
class Solution:
    def canJump(self, nums: list[int]) -> bool:
        goal: int = len(nums) - 1

        for i in range(len(nums) - 2, -1, -1):
            if nums[i] + i >= goal:
                goal = i

        return goal == 0


# @lc code=end
